src/source.cc

```diff
@@ -2120,7 +2121,25 @@ bool Source::View::on_button_press_event(GdkEventButton *event) {
     get_buffer()->select_range(start, end);
     return true;
   }
-  
+  if((event->type == GDK_BUTTON_PRESS) && (event->button == 3)){
+    auto menu=Menu::get().edit_menu;
```

Comment:
Actually, maybe two right click popup menus; one for when text is selected, and one for when text is not selected.